template_functions.go

```diff
@@ -231,6 +231,43 @@ func treeFunc(brain *Brain,
 
 }
 
+// byKey accepts a slice of KV pairs and returns a map of the top-level
+// key to all its subkeys. For example:
+//
+//		elasticsearch/a //=> "1"
+//		elasticsearch/b //=> "2"
+//		redis/a/b //=> "3"
+//
+// Passing the result from Consul through byTag would yield:
+//
+// 		map[string]map[string]string{
+//	  	"elasticsearch": &dep.KeyPair{"a": "1"}, &dep.KeyPair{"b": "2"},
+//			"redis": &dep.KeyPair{"a/b": "3"}
+//		}
+//
+// Note that the top-most key is stripped from the Key value. Keys that have no
+// prefix after stripping are removed from the list.
+func byKey(pairs []*dep.KeyPair) (map[string][]*dep.KeyPair, error) {
+	m := make(map[string][]*dep.KeyPair)
+	for _, pair := range pairs {
+		parts := strings.Split(pair.Key, "/")
+		top := parts[0]
+		key := strings.Join(parts[1:], "/")
```

Comment:
Hehe - similar to my comment above, it's not possible with split, but it's possible with splitn ironically:

```go
gore> a := strings.Split("", "/")
[]string{""}
gore> a[0]
""
gore> a[1]
panic: runtime error: index out of range

goroutine 1 [running]:
main.main()
	/var/folders/z8/dh6vp0m920sb8fs04s1pjdnm0000gn/T/039058848/gore_session.go:16 +0x12e
exit status 2
error: exit status 1
exit status 1
gore> a[1:]
[]string{}
```

as opposed to:

```go
gore> a := strings.SplitN("", "/", 2)
[]string{""}
gore> a[0]
""
gore> a[1]
panic: runtime error: index out of range

goroutine 1 [running]:
main.main()
	/var/folders/z8/dh6vp0m920sb8fs04s1pjdnm0000gn/T/015549103/gore_session.go:16 +0x13a
```

So, to get the second item in the list, we can't use `[1]`, which felt weird to me...